build-tools/packages/build-cli/src/commands/generate/manifest.ts

```diff
@@ -0,0 +1,226 @@
+		);
+
+		return successfulBuilds.length > 0
+			? (successfulBuilds[0].buildNumber as string)
+			: undefined;
+	} catch (error) {
+		log?.errorLog("Error fetching successful builds:", error);
+		return undefined;
+	}
+}
+
+/**
+ * Fetches the dev version number released in the build feed.
+ * @param buildNumber - The build number.
+ * @returns The dev version number if found, otherwise undefined.
+ */
+async function fetchDevVersionNumber(
+	authHeader: string,
+	organization: string,
+	project: string,
+	buildNumber: string,
+	log?: Logger,
+): Promise<string | undefined> {
+	const REGISTRY_URL = `https://pkgs.dev.azure.com/${organization}/${project}/_packaging/build/npm/registry/`;
+	try {
+		const response = await fetch(`${REGISTRY_URL}/${PACKAGE_NAME}`, {
+			headers: { Authorization: authHeader },
+		});
+		const data = await response.json();
+		const buildVersionKey = Object.keys(data.time).find((key) => key.includes(buildNumber));
+
+		if (buildVersionKey !== undefined) {
+			return buildVersionKey;
+		}
+
+		log?.log(`No version with build number ${buildNumber} found.`);
+		return undefined;
+	} catch (error: unknown) {
+		log?.errorLog("Error fetching dev version number:", error);
+		return undefined;
+	}
+}
+
+/**
+ * Generates a modified manifest file with the specified version number.
+ * @param version - The version number.
+ */
+async function generateManifestFileForDevReleases(
+	version: string,
+	log?: Logger,
+): Promise<string | undefined> {
+	try {
+		const releasesResponse = await fetch(GITHUB_RELEASE_URL);
+		const releases = await releasesResponse.json();
+
+		let manifestAsset;
+
+		for (const asset of releases[0].assets) {
+			const includesCaretJson: boolean = asset.name.includes(".caret.json");
+			if (includesCaretJson) {
+				manifestAsset = asset;
+				break;
+			}
+		}
+
+		if (Object.keys(manifestAsset).length > 0) {
+			const manifest_url_caret = manifestAsset.browser_download_url;
+			log?.log(`Downloading latest internal manifest: ${manifest_url_caret}`);
+
+			const manifestResponse = await fetch(manifest_url_caret);
+			const manifestData = await manifestResponse.buffer();
+			const manifest_filename = manifest_url_caret.slice(
+				// eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+				manifest_url_caret.lastIndexOf("/") + 1,
+			);
+			const new_manifest_filename = `fluid-framework-release-manifest.client.${version}.caret.json`;
+
+			await fsPromises.writeFile(manifest_filename, manifestData);
+			await fsPromises.rename(manifest_filename, new_manifest_filename);
+
+			const modifiedManifestBuffer = await fsPromises.readFile(new_manifest_filename);
+			const modifiedManifest = JSON.parse(modifiedManifestBuffer.toString());
+
+			for (const key of Object.keys(modifiedManifest)) {
+				const includesInternal: boolean = modifiedManifest[key].includes("internal");
+				if (includesInternal) {
+					modifiedManifest[key] = version;
+				}
+			}
```

Comment:
If you're going to parse the JSON, you should parse it into one of the report types that are already defined. E.g. https://github.com/microsoft/FluidFramework/blob/afe0efdfa25a010197c6cc6a94ff450f62814972/build-tools/packages/build-cli/src/lib/release.ts#L18-L27

Reply:
Similarly, if you're parsing anyway, I would parse the full report instead of the caret/simple one. Then you could re-use some more stuff like writeReport: https://github.com/microsoft/FluidFramework/blob/388ceac8e88ffcbab014ed576e326c9209a82921/build-tools/packages/build-cli/src/commands/release/report.ts#L665-L690